content/features/featuresDeepDive/frameGraph/frameGraphClassFramework/frameGraphClassOverview.md

```diff
@@ -0,0 +1,305 @@
+---
+title: Overview of the frame graph framework
+image:
+description: Learn all about the Babylon.js Frame Graph system.
+keywords: diving deeper, frame graph, rendering, node editor, overview
+---
+
+The frame graph class framework consists of several main classes, described below.
+
+## [FrameGraph](/typedoc/classes/babylon.framegraph)
+This is the main class, whose purpose is to allow you to build and execute a frame graph.
+
+### Main methods and properties
+* `addTask(task)`. Adds a task to the graph.
+* `addPass(name, whenTaskDisabled)`, `addRenderPass(name, whenTaskDisabled)`, `addObjectListPass(name, whenTaskDisabled)`. Methods that create a new pass for the currently processed task and return that pass. These methods can only be called from a `FrameGraphTask.record()` method, which is the method responsible for creating the passes of a task.
+* `build()`. Traverses all tasks in the graph and calls their `record()` method, which in turn will create the task's passes. This is also when the actual textures are allocated and linked to the texture handles created in the frame graph.
+* `execute()`. Traverses all tasks in the graph and executes the passes for each of them.
+* **textureManager**. This property gives you access to the frame graph's [Texture manager](#framegraphtexturemanager).
+* **optimizeTextureAllocation**. This property indicates that texture allocation should be optimized (i.e., reuse existing textures when possible to limit GPU memory usage).
+* **pausedExecution**. Indicates whether the execution of the frame graph is paused (default is false).
+
+You should generally disable frame graph execution before calling `await FrameGraph.whenReadyAsync()`, so that the frame graph is not executed by the main rendering loop before everything is ready, which could cause errors.
```

Comment:
I guess this is when user do modification on the framegraph (like adding a new pass) and wants to apply that changes? I'm missing the use case when I need to take care of that.